src/test/java/com/uid2/admin/vertx/SharingServiceTest.java

```diff
@@ -593,6 +593,38 @@ void KeysetSetBadWhitelistSiteId(Vertx vertx, VertxTestContext testContext) {
         });
     }
 
+    @Test
+    void KeysetSetIgnoresAlreadySetSitesWhenChecking(Vertx vertx, VertxTestContext testContext) {
+        fakeAuth(Role.ADMINISTRATOR);
+
+        mockSiteExistence(5, 22, 6);
+
+        Map<Integer, Keyset> keysets = new HashMap<Integer, Keyset>() {{
+            put(3, new Keyset(3, 7, "test", Set.of(12, 25), Instant.now().getEpochSecond(),true, true));
+            put(4, new Keyset(4, 4, "test", Set.of(5), Instant.now().getEpochSecond(),true, true));
+        }};
+
+        setKeysets(keysets);
+
+        String body = "  {\n" +
+                "    \"allowed_sites\": [\n" +
+                "      22,\n" +
+                "      25,\n" +
```

Comment:
should we add a site that doesn't exist? That was the scenario that we're fixing

Reply:
Yes this is what it is doing, 25 does not exist

Reply:
Added some comments